sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -58,6 +59,78 @@ def download_gcs_blob(local_path, gcs_path):
     blob = bucket.blob(gcs_blob)
     blob.download_to_filename(local_path)
 
+class DependencyVersion(object):
+  """ DependencyVersion specifies the versions """
+  def __init__(self, version=None, min_version=None, max_version=None):
+    """ if version is specified, no need for min_version or max_version;
+     if both are specified, version is adopted """
+    if version is not None:
+      self._min_version = version
+      self._max_version = version
+    else:
+      self._min_version = min_version
+      self._max_version = max_version
+
+  @property
+  def min_version(self):
+    return self._min_version
+
+  @min_version.setter
+  def min_version(self, min_version):
+    self._min_version = min_version
+
+  def has_min_version(self):
+    return self._min_version != None
+
+  @property
+  def max_version(self):
+    return self._max_version
+
+  @max_version.setter
+  def max_version(self, max_version):
+    self._max_version = max_version
+
+  def has_max_version(self):
+    return self._max_version != None
+
+  def has_versions(self):
+    return (self.has_min_version()) or (self.has_max_version())
+
```

Comment:
Let's add `__str__` or `__repr__` that outputs this like a line in requirements.txt

Reply:
I don't think this is quite a good approach because the class versionedDependency should be self contained as an information aggregator for versioned packages instead of being used for requirements.txt specifically. For general code reuse, it might be better to leave the functionality as it should be. WDYT?

Reply:
Yes, it can be used in other contexts, but having a good `__str__` or `__repr__` is a benefit.

Reply:
__str__ and __repr__ might be useful for print statements. Extra codes need to be written to generate customized strings, for example the requirement file for pip. 
How about adding __str__ and __repr__ for simple format, such as "package name: tensorflow, min version: 0.1.0, max version: 0.2.0"?